### PR TITLE
Remove unneeded string operations

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/12477/6/1

<details>


</details>

File: test/csync/vio_tests/check_vio_ext.cpp [hunk 1]
```diff
@@ -45,8 +45,8 @@ int oc_mkdir(const QString &path)
 static mbchar_t wd_buffer[WD_BUFFER_SIZE];
 
 typedef struct {
-    QByteArray result;
```
Comment (TheOneRing) [inline test/csync/vio_tests/check_vio_ext.cpp:48]: Not sure about the value of that unit test by now.....

**Comment by saw-jan** [comment]
CI stopped with the following message:
```bash
starting tests
11:21:42:280 Info: Starting server of Squish version 6.7.3 Alpha.
11:21:42:288 Info: Listening on 0.0.0.0:4322
11:21:42:288 Info: Listening on ws://127.0.0.1:45009?zWKcD9u1ONE8SoLEWr_N
11:21:42:288 Info: Announcing server presence via SSDP 
```
Tests are not running. Restarted CI.

File: src/csync/vio/csync_vio_local_win.cpp [hunk 2]
```diff
@@ -133,16 +131,16 @@ std::unique_ptr<csync_file_stat_t> csync_vio_local_readdir(csync_vio_handle_t *h
...
+  const QString path = QString::fromWCharArray(handle->ffd.cFileName);
   if (path == QLatin1String(".") || path == QLatin1String(".."))
       return csync_vio_local_readdir(handle, vfs);
 
   file_stat.reset(new csync_file_stat_t);
-  file_stat->path = path.toUtf8();
+  file_stat->path = path;
 
-    if (vfs && vfs->statTypeVirtualFile(file_stat.get(), &handle->ffd)) {
+  if (vfs && vfs->statTypeVirtualFile(file_stat.get(), &handle->ffd)) {
```
Comment (fmoc) [inline src/csync/vio/csync_vio_local_win.cpp:141]: The indentation is more off than it used to be. Please revise and if necessary fix for the entire file once and for all.

**Review by fmoc — CHANGES_REQUESTED** [review]

**Review by fmoc — APPROVED** [review]
Indentation mess will be fixed in follow-up PR.

**Pull request merged**